src/hooks/use-query.tsx

```diff
@@ -0,0 +1,41 @@
+import {
+  useQuery as useReactQuery,
+  UseQueryOptions
+} from '@tanstack/react-query'
+import { AxiosError } from 'axios'
+
+interface UseQueryProps<
+  TQueryFnData,
+  TError,
+  TData,
+  TQueryKey extends unknown[]
+> {
+  queryKey: TQueryKey
+  queryFn: () => Promise<TQueryFnData>
```

Comment:
I don't think that `queryFn` should return Promise with type of `QueryFnData` (or it's very confusing). Try to add better type names, that will be understandable without going to tanstack docs)
https://github.com/TanStack/query/discussions/1477